gravatar/openapi/templates/data_class.mustache

```diff
@@ -60,14 +61,14 @@ import {{packageName}}.infrastructure.ITransformForStorage
 @Deprecated(message = "This schema is deprecated.")
 {{/isDeprecated}}
 {{>additionalModelTypeAnnotations}}
-{{#nonPublicApi}}internal {{/nonPublicApi}}{{#discriminator}}interface{{/discriminator}}{{^discriminator}}public data class{{/discriminator}} {{classname}}{{^discriminator}} (
+{{#nonPublicApi}}internal {{/nonPublicApi}}{{#discriminator}}interface{{/discriminator}}{{^discriminator}}public class{{/discriminator}} {{classname}}{{^discriminator}} private constructor(
```

Comment:
Curious but the `public` keyword has no condition so if we use `nonPublicApi` we will end up with `internal public class`, right?

Reply:
Good point. 

I've added the condition, but to be honest, I am not 100% sure that the kotlin generator supports it. I don't see it on the [kotlin docs](https://openapi-generator.tech/docs/generators/kotlin), and I couldn't use it🤷🏽‍♂️. Anyway, it looks better with the condition, just in case.